packages/nimble-components/src/rich-text-mention/base/models/mention-internals.ts

```diff
@@ -63,6 +69,7 @@ export class MentionInternals {
         this.icon = options.icon;
```

Comment:
Instead of defining this upstream, we should change mentionInternals to be the iconTemplate. The iconTemplate instance can and should be shared instead of reconstructed everytime the configuration changes.

i.e.

```suggestion
        this.iconTemplate = html`<${options.icon} slot="start"></${options.icon}>`;
```

Reply:
Updated as per the suggestion.